numba/core/controlflow.py

```diff
@@ -583,13 +588,17 @@ def _find_back_edges(self):
         succs_state = {}
         entry_point = self.entry_point()
 
+        checked = set()
+
         def push_state(node):
             stack.append(node)
             succs_state[node] = [dest for dest in self._succs[node]]
 
         push_state(entry_point)
 
         while stack:
+            if stats is not None:
+                stats['iteration_count'] += 1
```

Comment:
TODO: lift this branch out of the loop and just do a simple counter increment in the loop